accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -540,6 +545,18 @@ all non-extension type members named `n` that _DV_ has, again using a
 repeated computation of the instantiated type of each superinterface on the
 path to the given non-extension type superinterface.
 
+There are two exceptions to these rules: _DV_ does not have a setter named
```

Comment:
> We also need to consider an inherited `n=` and `n()` a conflict.

That's already true: In this case _DV_ 'has' both members, and that's an error according to the normal class member conflict rules. The exceptions that I added are there to _cancel_ the error in the special case where a method is declared in _DV_ and a setter is "inherited", or vice versa, because we just don't "inherit" the one from the superinterface. Those exceptions do not make any difference when both are "inherited", we just get the same error now that we got previously.

> has a unique super-extension type

(.. right, 'unique' is only used to ensure that it is well-defined to talk about _the_ declaration that _DV_ has; however, _DV_ has a compile-time error if it isn't unique, and we don't really care whether _DV_ has or doesn't have something when it's an error ..)

> The "In both cases" seems spurious

Why do you think so? One case is when _DV_ declares a member named `n` and the other case is when _DV_ "inherits" a member named `n`. Which one of those is not needed?

I think it is necessary to be able to say exactly which piece of code is executed in the case where a receiver `r` whose type is the type introduced by _DV_ (possibly with some actual type arguments) occurs in a member access like `r.n` or `r.n()`. That should be well-defined both when _DV_ declares a member named `n`, and when _DV_ "inherits" such a member from some superinterface.

It is true that 'when this is unique' is relevant to the case where the declaration is "inherited" and irrelevant to the case where _DV_ declares that member (because that one is always unique, relying on the general rule that you mention, against multiple declarations with the same name in the same scope).

We could fiddle with the words in order to avoid saying 'if unique' about the case where that's trivially true. Do you think the result would be less complex?

"Here's the cheetah, the antelope, and the turtle, and all of them must proceed with less than 25 mph" vs. "Here's the cheetah, the antelope, and the turtle, and all of them except the turtle must proceed with less than 25 mph". Does it help?

> unique (meaning only one superinterface has a member, or all superinterfaces have the same member, through diamond inheritance?)

It is a compile-time error if _DV_ has multiple extension type member declarations with the same name (so diamond inheritance is fine, the question is whether we have multiple declarations or not, and that's the reason why I maintain that _DV_ "has" a declaration that occurs in some other extension type declaration than _DV_).

It is not an error if _DV_ has multiple non-extension type members with the same name (they must all denote the same member implementation), but it's an error if the combined member signature doesn't exist.

Finally, it's a compile-time error if _DV_ has an extension type member as well as a non-extension type member with the same name.

> if the inherited declaration is not unique .. then _DV_ just doesn't have an extension type member named `n`?

I believe we're talking about extension-type members here (that is, members declared in extension type declarations). If _DV_ has multiple declarations named `n` then it's an error. So we don't really care whether _DV_ does or does not have anything, but the text arguably says that _DV_ has such a member.

We just make it explicit that we cannot talk about "the" member declaration named `n` that _DV_ has, because that's not a well-defined concept.

Reply:
I clarified the point about uniqueness at the location where 'precludes' is defined.